src/util/psbt/responsibilities.rs

```diff
@@ -0,0 +1,147 @@
+// Rust Bitcoin Library
+// Written in 2020 by
+//   The Rust Bitcoin developers
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Implementation of PSBT responsibilities according to BIP-174
+
+use std::marker::PhantomData;
+use std::error::Error;
+use std::fmt::{Formatter, Display};
+use secp256k1::{Message, Signature};
+use util::bip32::{Fingerprint, DerivationPath};
+use super::PartiallySignedTransaction;
+
+/// Trait for a generic PSBT responsibility (decorator+facade)
+pub trait Responsibility where Self: Sized {
```

Comment:
Is this responsibility abstraction actually useful? What's the intended use case?

Reply:
There might be functions taking PSBT from a responsibility and internally extracting from it - or packing into it.